Fix -Werror=calloc-transposed-args error in `url.c` with gcc 14.

```
src/url.c: In function 'curl_url':
src/url.c:1026:24: error: 'calloc' sizes specified with 'sizeof' in the earlier
argument and not in the later argument [-Werror=calloc-transposed-args]
 1026 |   return calloc(sizeof(struct Curl_URL), 1);
      |                        ^~~~~~
src/url.c:1026:24: note: earlier argument should specify number of elements,
later size of each element
src/url.c: In function 'curl_url_dup':
src/url.c:1046:38: error: 'calloc' sizes specified with 'sizeof' in the earlier
argument and not in the later argument [-Werror=calloc-transposed-args]
 1046 |   struct Curl_URL *u = calloc(sizeof(struct Curl_URL), 1);
      |                                      ^~~~~~
src/url.c:1046:38: note: earlier argument should specify number of elements,
later size of each element
```

--- gmni-1.0-origin/src/url.c
+++ gmni-1.0/src/url.c
@@ -1023,7 +1023,7 @@
  */
 struct Curl_URL *curl_url(void)
 {
-  return calloc(sizeof(struct Curl_URL), 1);
+  return calloc(1, sizeof(struct Curl_URL));
 }

 void curl_url_cleanup(struct Curl_URL *u)
@@ -1043,7 +1043,7 @@

 struct Curl_URL *curl_url_dup(struct Curl_URL *in)
 {
-  struct Curl_URL *u = calloc(sizeof(struct Curl_URL), 1);
+  struct Curl_URL *u = calloc(1, sizeof(struct Curl_URL));
   if(u) {
     DUP(u, in, scheme);
     DUP(u, in, user);
